src/JsonLdWriter.php

```diff
@@ -215,10 +228,64 @@ protected function extractPathFromUrl($original)
         return str_replace("/", "", $parts["path"]);
     }
 
-    protected function convertImageUrlToBinary(string $url): ?string
+    protected function convertImageToBinary(array|string $value): mixed
     {
-        // Add code to extract and convert image
+        // First check if we have a simple url string, an array of images or an ImageObject.
+        if (is_string($value)) {
+            // Simple String that we can convert.
+            return $this->convertUrlToBinary($value) ?? $value;
+        } elseif (is_array($value)) {
+            // Either an ImageObject or an Array of images/ImageObjects.
+            if (array_key_exists('@type', $value)) {
+                // ImageObject, we need to find out which field is filled.
+                return $this->convertImageObjectUrlToBinary($value);
+            } else {
+                // Array of images, so we just call this method on every element.
```

Comment:
Changed it so that it grabs the first element with [`array_shift()`](https://www.php.net/manual/en/function.array-shift.php), converts it and uses [`array_unshift()`](https://www.php.net/manual/en/function.array-unshift.php) to push it back to the start.